kedro/runner/runner.py

```diff
@@ -232,15 +243,16 @@ def _collect_inputs_from_hook(
     )
 
     additional_inputs = {}
-    for response in hook_response:
-        if response is not None and not isinstance(response, dict):
-            response_type = type(response).__name__
-            raise TypeError(
-                f"`before_node_run` must return either None or a dictionary mapping "
-                f"dataset names to updated values, got `{response_type}` instead."
-            )
-        response = response or {}
-        additional_inputs.update(response)
+    if hook_response is not None:
```

Comment:
Yep, without the condition I was getting the `TypeError: 'NoneType' object is not iterable` error; from the implementation I am unsure on how None was successfully handled previously, as there are no checks before attempting to iterate through - is there a previous example of this? 

`print(hook_response)` simply returned `None`. Additionally, I believe that if `[None]` was returned instead the check would no longer be necessary, as the condition on the following line (currently 239) would catch it.

Reply:
@AntonyMilneQB "Does calling hook_manager.hook.before_node_run with such a hook not actually return None but instead [None] or something?" Yes it does indeed return an empty list instead of None. 

So with this implementation we return None because of:

```
def __call__(self, *args, **kwargs):
        pass
``` 
in the `_NullPluginManager`

If you change that to:

```
def __call__(self, *args, **kwargs):
        return []
```

you don't need that if condition.

Reply:
I would keep the if condition in place and just add a comment to it to clarify why it's needed.